### PR TITLE
Changes for OpenSAML3 migration

### LINKED ISSUE #7156: Migrate the version of OpenSAML from OpenSAML2 to OpenSAML3
### Description
Since OpenSAML2 reached its end of life 3 years back ([July 31, 2016](https://wiki.shibboleth.net/confluence/display/OpenSAML/Home)) for Java. Security fixes are no longer supported due to the situation we need to upgrade it to OpenSAML3 ASAP. The process involves conversion of the available repositories which are using OpenSAML2 to be converted to OpenSAML3. 

### The major changes from OpenSAML2 to OpenSAML3 includes:

- The structure of the OpenSAML dependencies has changed between version 2 and 3. OpenSAML 3.x is structured as a Maven multi-module project.
-  The Message Context has changed to become more modular. `MessageHandlers` has been added to process the messages using message context.
-  The syntax has been changed for some processes:
    - The syntax for using the `HTTPRedirectDeflateEncoder` has changed to a small degree, mainly due to the new design of the message context.
    - The syntax for sending SOAP messages has changed substantially.
- A lot of package names have changed from version 2 to 3.
- Some of the classes has been renamed for example:
   - The `EntityIdCriteria` class used for resolving credentials is renamed to `EntityIdCriterion`.
   - The `EncryptionParameters` class used for specifying encryption options is renamed to `DataEncryptionParameters`.
- Some methods are dropped out or either moved into another class. For example:
   - The `Configuration.validateNonSunJAXP` method has been removed as it is not longer necessary to validate this. 
   - `Configuration.validateJCEProviders` method has been replaced by the `init()` method of `JavaCryptoValidationInitialize`.
- Changes to Field names such as:
   - The `StatusCode.SUCCESS_URI` used in the Artifact Response has been renamed to `StatusCode.SUCCESS`.

### Resources for the migration
- https://wiki.shibboleth.net/confluence/display/OpenSAML/OSTwoDeveloperManual
- https://wiki.shibboleth.net/confluence/display/OS30/Home
- https://build.shibboleth.net/nexus/content/sites/site/java-opensaml/3.4.2/apidocs/
- https://nxp.evelopers.com/content/43dfe113-26c9-46c7-97b7-107c735a5073/?attach=false&language=&url=
- http://shibboleth.1660669.n2.nabble.com/Shibboleth-Developers-f1660781.html

### Suggested Labels
Type/Improvement, Priority/High, Component/SAML, Complexity/High

### Suggested Assignee
@deshankoswatte

### PR DESCRIPTION
**Pull request by deshankoswatte**
### Proposed changes in this pull request
Changes for OpenSAML3 migration

### When should this PR be merged
Not specified yet

### Prerequisites
- wso2/orbit#395 - Orbit for OpenSAML3
- wso2/carbon-identity-framework#2473 - Migration from OpenSAML 2.x to 3.x

Fixes: wso2/product-is#7156

### CODE REVIEW

File: components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/security/X509CredentialImpl.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ public X509Certificate getEntityCertificate() {
     }
 
     @Override
-    public CredentialContextSet getCredentalContextSet() {
+    public CredentialContextSet getCredentialContextSet() {
         return null;
     }
```
Comment (deshankoswatte) [inline components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/security/X509CredentialImpl.java]: Change the method to:
```
/***
 * get a credential context set
 * @return Method not supported so will return null
 */
public CredentialContextSet getCredentialContextSet() {
      return null;
}
```

File: components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/util/SSOAgentUtils.java [hunk 2]
```diff
@@ -119,10 +119,10 @@ public static String createID() {
...
-                DefaultBootstrap.bootstrap();
+                SAMLInitializer.doBootstrap();
                 isBootStrapped = true;
-            } catch (ConfigurationException e) {
-                throw new SSOAgentException("Error in bootstrapping the OpenSAML2 library", e);
+            } catch (InitializationException e) {
+                throw new SSOAgentException("Error in bootstrapping the OpenSAML3 library", e);
             }
         }
     }
```
Comment (deshankoswatte) [inline components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/util/SSOAgentUtils.java]: Add space after method signature

Comment (deshankoswatte) [reply components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/util/SSOAgentUtils.java]: Fixed

File: components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/security/X509CredentialImpl.java (hunk 1 above)
Comment (deshankoswatte) [reply components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/security/X509CredentialImpl.java]: Fixed

**Review by GDRDABARERA — APPROVED** [review]

**Review by madurangasiriwardena — APPROVED** [review]

**Pull request merged**